shakenfist/external_api/app.py

```diff
@@ -607,8 +607,9 @@ def post(self, name=None, cpus=None, memory=None, network=None,
              instance_uuid=None):
         global SCHEDULER
 
-        # We need to sanitise the name so its safe for DNS
-        name = re.sub(r'([^a-zA-Z0-9_\-])', '', name)
+        # Check that the instance name is safe for use as a DNS host name
+        if name != re.sub(r'([^a-zA-Z0-9_\-])', '', name) or len(name) > 63:
+            return error(400, "Instance name must be useable as a DNS host name")
```

Comment:
Quotes! I see quotes!

Reply:
Rebased to current master and fixed the quotes